src/article_id.rs

```diff
@@ -336,20 +356,53 @@ impl<'a> TryFrom<&'a str> for ArticleId<'a> {
 			return Err(ExpectedBeginningLiteral);
 		}
 
-		let inner_parts: Vec<&str> = parts[1].split(ArticleId::TOKEN_DOT).collect();
-		if inner_parts.len() != 2 {
-			return Err(ExpectedNumberVv);
+		Self::try_from_id(parts[1])
+	}
+}
+
+#[derive(PartialEq, Eq, Debug)]
+pub enum UrlParsingError<'a> {
+	MissingDomain,
+	InvalidDomain(&'a str),
+	MissingUrlSegments,
+	MissingFormat,
+	InvalidArxivFormat,
+	MissingArticleId,
+	InvalidArticleId(ArticleIdError),
+}
+
+#[cfg(feature = "url")]
+#[cfg_attr(docsrs, doc(cfg(feature = "url")))]
+impl<'a> TryFrom<&'a url::Url> for ArticleId<'a> {
+	type Error = UrlParsingError<'a>;
+	fn try_from(value: &'a url::Url) -> Result<Self, Self::Error> {
```

Comment:
This function should have a documentation block with a one-liner description and a small section of examples